assets/app/scripts/controllers/newfromtemplate.js

```diff
@@ -0,0 +1,161 @@
+      var redirect = URI('/error').query({
+      	error_description: message,
+      }).toString();
+      $location.url(redirect);
+    }
+    
+    var dcContainers = $parse('template.controllerTemplate.podTemplate.desiredState.manifest.containers');
+    var stiBuilderImage = $parse('parameters.strategy.stiStrategy.image');
+    var outputImage = $parse('parameters.output.to.name || parameters.output.DockerImageReference');
+    
+    function deploymentConfigImages(dc) {
+      var images = [];
+      var containers = dcContainers(dc);
+      if (containers) {
+        containers.forEach(function(container) {
+          images.push(container.image);
+        });
+      }
+      return images;
+    }
+
+    function imageItems(data) {
+      var images = [];
+      var dcImages = [];
+      var outputImages = {};
+      data.items.forEach(function(item) {
+        if (item.kind == "BuildConfig") {
+          var builder = stiBuilderImage(item)
+          if(builder) {
+            images.push({ name: builder });
+          }
+          var output = outputImage(item);
+          if (output) {
+            outputImages[output] = true;
+          }
+        }
+        if (item.kind == "DeploymentConfig") {
+          dcImages = dcImages.concat(deploymentConfigImages(item));
+        }
+      });
+      dcImages.forEach(function(image) {
+        if (!outputImages[image]) {
+          images.push({ name: image });
+        }
+      });
+      return images;
+    }
+    
+    $scope.createFromTemplate = function() { 
+      DataService.create("templateConfigs", $scope.template, $scope).then(
+        function(config) { // success
+          var titles = { 
+            started: "Creating " + $scope.template.metadata.name + " in project " + $scope.projectName,
+            success: "Created " + $scope.template.metadata.name + " in project " + $scope.projectName,
+            failure: "Failed to create " + $scope.template.metadata.name + " in project " + $scope.projectName
+          };
+          // TODO: Determine how to generate Help Links - would they be annotations on the template?
+          var helpLinks = [ { title:"Additional Information", link: "https://github.com/openshift/origin" } ];
+          TaskList.add(titles, helpLinks, function() {
+            var d = $q.defer()
+            DataService.createList(config.items, $scope).then(
+              function(result) {
+                var alerts = [];
+                var hasErrors = false;
+                if (result.failure.length > 0) {
+                  result.failure.forEach(
+                    function(failure) {
+                      var objectName = "";
+                      if (failure.data && failure.data.details) {
+                        objectName = failure.data.details.kind + " " + failure.data.details.id;
+                      }
+                      alerts.push({ type: "error", message: "Cannot create " + objectName, details: failure.data.message })
```

Comment:
We should put a TODO here that we should use the causes returned from the API to provide more specific reasons for why the failure happened